src/main/java/org/team1540/robot2023/utils/PoseMedianFilter.java

```diff
@@ -38,8 +38,15 @@ private double median(LinkedList<Double> list) {
             return (copyList.get(copyList.size()/2)+copyList.get(copyList.size()/2-1))/2;
         }
     }
-    public Translation2d medianPose() {
-        return new Translation2d(median(xVals), median(yVals));
+    public Optional<Pose2d> checkOutlier(Pose2d pose) {
```

Comment:
Interesting choice to use Optional rather than just returning null, since checking whether this is empty and checking whether something is null is essentially the same.